### DISCUSSION TITLE
What is the reason for docutils & jinja version restrictions?

### DISCUSSION THREAD

**Discussion by mgorny** [executablebooks/meta#762] [Q&A]
Hi,

I've noticed that the newest release claims not to support `docutils~=0.18.0` nor `jinja2>=3.1.0`. However, I've just run the test suite against `docutils==0.18.1` and `jinja2==3.1.2`, and all tests passed. Is there some untested hidden breakage or are the version restrictions unnecessarily restrictive?

**Comment by pradyunsg** [comment]
For docutils: Each minor release introduces markup changes, which won't result in a broken build and don't show up as a failure, but the rendered output is different and likely isn't stylised correctly.

**Reply by rossbar** [reply]
As a concrete example, we experienced problems with collapsible elements going from docutils 0.16 to 0.17 - see numpy/user-survey-2020-details#36.

**Reply by choldgraf** [reply]
Also just to note, the ~= means that patch versions should still be supported. It is only the next minor version release of docutils that will need an explicit version bump here. Minor version releases with docutils are often breaking in some form or another.

For Jinja if I recall, there was a bug that caused later version to break so maybe that is why? I'm not sure on that one though.

**Reply by pradyunsg** [reply]
I've written some words about conflating "the build is successful in our build system and tests pass" with "it works correctly" -- in the context of Sphinx themes.

https://github.com/pradyunsg/sphinx-theme-builder/discussions/31#discussioncomment-3025601
